### ISSUE TITLE
M.2 NGFF Key B+M Adapter - WiFi OK, no Bluetooth from BCM94360CS2

### ISSUE THREAD

**Issue by naryfa** [sarkrui/NUC8i7BEH#27]
Hi,

So the guide mentions this:

> _BCM94360CS2 (2 antenna IPEX4 connector) Wi-Fi/Bluetooth + M.2 NGFF Key B+M Adapter. Intel antenna replacement is NOT required but 2.4 GHz will have conflicts with bluetooth as they share the same connector._

I used the BCM94360CS2 with the onboard Intel IPEX4/MHF4 connectors. My question is, **would it completely stop BT from showing up in macOS, or do I just have a bad adapter?** I mean, I get the thing with the antenna and conflicts, but here the OS appears to not even detect the hardware. (If I remember correctly, trying BCM943602CDP in the past yielded similar results).

All I want is that the AirDrop would function 😥

Screenshots for reference:

![Screen Shot 2021-03-06 at 11 45 04 AM](https://user-images.githubusercontent.com/38587586/110214475-aedca780-7e72-11eb-9af7-0883639b4d61.png)
![Screen Shot 2021-03-06 at 11 45 20 AM](https://user-images.githubusercontent.com/38587586/110214476-aedca780-7e72-11eb-813e-d9a2e7f500e3.png)

**Comment by naryfa** [comment]
Update:

I just tried a WiFi card from another MacBook Air. Same thing, no BT showed up.

**Comment by sarkrui** [comment]
Hi,

1. You can first try installing Windows and see if Bluetooth works (Drivers are required. [download link](https://github.com/sarkrui/Hackintosh-Z390-Aorus-Pro-9700K-RX580/tree/master/(Windows)%20BCM94360:94331%20Drivers) )
2. You can try rebuild the kext cache with the below command.

```
sudo kextcache -i /
```

Good luck

**Comment by naryfa** [comment]
Thank you for response.

Great idea, I'll try number 1 tomorrow, will have the time.

As for the cache, I did it many times (just today I added the microSD card reader kext, after which hackintool rebuilt the cache on its own).

**I assume I don't need any kexts, as BCM94360CS2 is a native card, correct?**



P.S. I checked the wire continuity (the red and black wires from the adapter to USB header) and they're fine. I also ordered two more adapters. If these don't work with any of the three native cards I've got, then it's the USB header that's toast.
